### PR TITLE
[SRVCOM-2658] Add Serverless 1.31 release notes

### PR DESCRIPTION
**Pull request by rh-max**
Version(s):
`serverless-docs-1.30`+

Issue:
https://issues.redhat.com/browse/SRVCOM-2658

Link to docs preview:
1. Serverless 1.31.0 release notes:
https://68620--docspreview.netlify.app/openshift-serverless/latest/about/serverless-release-notes#serverless-rn-1-31-0_serverless-release-notes
2. Generally Available and Technology Preview features table:
https://68620--docspreview.netlify.app/openshift-serverless/latest/about/serverless-release-notes#serverless-tech-preview-features_serverless-release-notes
3. Deprecated and removed features table:
https://68620--docspreview.netlify.app/openshift-serverless/latest/about/serverless-release-notes#serverless-deprecated-removed-features_serverless-release-notes

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

File: modules/serverless-rn-1-31-0.adoc
Comment (mgencur) [inline modules/serverless-rn-1-31-0.adoc:26]: Where does this change come from?
I'm not aware of this and looking at deployments from 1.31, I see this:
```
oc get pods -n knative-serving | grep -e domain -e controller
controller-66658b8885-ss5jq                                   2/2     Running     0          4m2s
controller-66658b8885-vxtgt                                   2/2     Running     0          4m19s
domain-mapping-56c6d67d66-82ljb                               2/2     Running     0          4m25s
domain-mapping-56c6d67d66-xjj26                               2/2     Running     0          4m25s
domainmapping-webhook-86c5f8d4df-b4s89                        2/2     Running     0          4m25s
domainmapping-webhook-86c5f8d4df-mk5tj                        2/2     Running     0          4m25s

```

File: modules/serverless-rn-1-31-0.adoc
Comment (mgencur) [inline modules/serverless-rn-1-31-0.adoc:37]: It was this one: https://issues.redhat.com/browse/SRVKE-1510
But the current text is misleading, we should add at the beginning that PingSource metrics were unavailable when ServiceMesh integration in full mesh was used with STRICT PeerAuthentication.

File: modules/serverless-rn-1-31-0.adoc
Comment (mgencur) [inline modules/serverless-rn-1-31-0.adoc:42]: Hmm, this note should NOT be under Known issues, IMO.

File: modules/serverless-tech-preview-features.adoc
Comment (mgencur) [inline modules/serverless-tech-preview-features.adoc:25]: We're now mentioning that:
* Eventing integration with Red Hat OpenShift Service Mesh - TP
* Service Mesh mTLS - GA

Where does the integration with Service Mesh being TP come from? I know we have one other part - Multi-tenancy with Service Mesh which is currently TP.

**Comment by briandooley** [comment]
@nainaz Can you give this a look?

File: modules/serverless-rn-1-31-0.adoc
Comment (rh-max) [reply modules/serverless-rn-1-31-0.adoc:26]: @mgencur https://issues.redhat.com/browse/SRVKS-1155

File: modules/serverless-rn-1-31-0.adoc
Comment (rh-max) [reply modules/serverless-rn-1-31-0.adoc:42]: Moved to "New features" (that's where we put updates of this type).

File: modules/serverless-rn-1-31-0.adoc
Comment (mgencur) [reply modules/serverless-rn-1-31-0.adoc:26]: I see this https://github.com/knative/serving/pull/14024 and it was rejected, then I see this commit merged only in release-v1.11 branch which is Serverless 1.32: https://github.com/knative/serving/commit/fc166ac8b459a73116ca5535e168aba2d97a2626 ; so only the domainmapping-webhook is merged with the other webhook and that will be available in 1.32.
Let's remove the note from the release notes for now.

File: modules/serverless-tech-preview-features.adoc
Comment (mgencur) [reply modules/serverless-tech-preview-features.adoc:30]: * Using Service Mesh to isolate network traffic with OpenShift Serverless - it was not available at all in 1.29, 1.30. And it is TP in 1.31 for the first time.
* Service Mesh mTLS
  https://issues.redhat.com/browse/SRVKE-1410 - Developer Preview in 1.29
  https://issues.redhat.com/browse/SRVKE-1473 - Technical Preview in 1.30
  https://issues.redhat.com/browse/SRVKE-1450 - GA in 1.31
* Eventing integration with Red Hat OpenShift Service Mesh - Not sure if we want to mention this at all. It's an umbrella for both the above issues so I'm not sure if we want to say it's GA or TP. https://issues.redhat.com/browse/SRVKE-1311 but

**Comment by mgencur** [comment]
LGTM

**Review by skopacz1 — APPROVED** [review]
Looks good to me! Thanks for the extra detailed preview links. Also, be sure to squash your commits into one before starting merge review.

**Comment by rh-max** [comment]
@mgencur Could you PTAL at these new entries suggested by Naina? https://github.com/openshift/openshift-docs/pull/68620/commits/f2e1157ba67e81e42f20342198440879130a7008

@skopacz1 Same question, also only about https://github.com/openshift/openshift-docs/pull/68620/commits/f2e1157ba67e81e42f20342198440879130a7008

Thanks.

File: modules/serverless-rn-1-31-0.adoc
Comment (jeana-redhat) [inline modules/serverless-rn-1-31-0.adoc:37]: Is there a noun that can follow `PodDistruptionBudget` and `HorizontalPodAutoscaler`? Not sure if "object" applies or something else

File: modules/serverless-rn-1-31-0.adoc
Comment (jeana-redhat) [inline modules/serverless-rn-1-31-0.adoc:50]: Is there a noun that can follow `knative-eventing`? Similarly, should STRICT and PeerAuthentication be in backticks with descriptions of what they are (rather than being used as plain text)?

**Review by jeana-redhat — COMMENTED** [review]
Overall LGTM, just left a few questions about some object name type things. If those can be labelled, please do so. When you're ready for merge, please squash commits :slightly_smiling_face: 

/remove-label merge-review-in-progress
/remove-label merge-review-needed

**Review by jeana-redhat — APPROVED** [review]
Awesome, LGTM

/remove-label merge-review-needed

edit to say: will merge and pick as soon as Travis finishes....

**Comment by jeana-redhat** [comment]
@rh-max oddly, this will cherrypick cleanly into 1.30 but not 1.31. I am going to hold off on merging the 1.30 version until things are working right, can you look into the 1.31 issue?

**Pull request merged**
